backend/common/src/main/java/ai/verta/modeldb/common/futures/Future.java

```diff
@@ -0,0 +1,387 @@
+package ai.verta.modeldb.common.futures;
+
+import ai.verta.modeldb.common.exceptions.ModelDBException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.function.*;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class Future<T> {
+  /**
+   * This instance is the global executor for all Futures. Everything will fail if it is not set
+   * properly.
+   */
```

Comment:
Should we set `@NonNull` or something like that? Not sure how that annotation works in java, but maybe it helps?

Reply:
Setting that unfortunately won't help, since it won't be detected at compile time as an issue (which is the only place that annotation is ever used).